vng/base/scrap_functions/license_number_with_company_name.py

```diff
@@ -71,42 +63,59 @@ def detectLicensePlate(self, image):
                     bbox = box  # (center_x, center_y, width, height)
                     if flag == True:
                         r_box = [int(round(item, 0)) for item in bbox]
-
                         offset = 10
                         im1 = image.crop((bbox[0] - offset, bbox[1] - offset, bbox[2] + offset, bbox[3] + offset))
                         image_path = 'detected.jpg'
                         im1.save(image_path)
                         img = cv2.imread(image_path, 1)
-                        img = cv2.resize(img, None, fx=3, fy=3, interpolation=cv2.INTER_CUBIC)
+                        img = cv2.resize(img, None, fx=10, fy=10, interpolation=cv2.INTER_CUBIC)
+                        # (thresh, img) = cv2.threshold(img, 85, 255, cv2.THRESH_BINARY)
                         cv2.imwrite(image_path, img)
-                        license_number = ''
+                        globals()['license_number'] = "" 
+                        license_numberr = ''
                         result = (self.model_dict["ocr_model"]).ocr(image_path, cls=True)
+                        # print(result)
                         ic = image.crop(r_box)
-                        for i in range(10):  # with the BLUR filter, you can blur a few times to get the effect you're seeking
+                        for i in range(30):  # with the BLUR filter, you can blur a few times to get the effect you're seeking
                             ic = ic.filter(ImageFilter.BLUR)
                         image.paste(ic, r_box)
-                        for line in result:
-                            license_number = license_number + " " + line[1][0]
-                        plates.append(license_number.strip())
-                        #print(f"Vehicle Licence Number:  {license_number}\n\n")
-                    #else:
-                       # print('No license number is detected')
+                        license_numberr = str(result[0][1][0])
+                        # for line in result:
+                            # if len(line)>0:
+                            #     print(line)
+                            #     license_number = license_number + " " + str(line[1][0])
+                        if license_numberr!= '':
+                            plates.append(license_numberr.strip())
+                            print(f"Vehicle Licence Number:  {license_numberr}\n\n")
+                        else:
+                            print('No license number is detected')
+                    else:
+                       print('No license number is detected')
+                    globals()['license_number'] = license_numberr
```

Comment:
working with globals is not a good idea.

Reply:
It is resoved now.